src/SpaceStationType.h

```diff
@@ -19,14 +19,60 @@ namespace SceneGraph {
 	class Model;
 }
 
+enum class DockStage { // <enum scope='DockStage' name=DockStage public>
+
+	NONE,
+	MANUAL,
+
+	DOCK_STAGES_BEGIN,
+
+	CLEARANCE_GRANTED,
+
+	DOCK_ANIMATION_NONE,
+	DOCK_ANIMATION_1,
+	DOCK_ANIMATION_2,
+	DOCK_ANIMATION_3,
+	DOCK_ANIMATION_MAX,
```

Comment:
Have you considered replacing the hard-coded 4-stage docking animation to instead be a dynamic list of Tags generated from the model file setup? I'd expect that to be significantly more flexible for future changes to the docking procedure (e.g. ships "flying themselves" along the docking path) and also allow e.g. using a "locator beacon" well outside the station for the first approach node.

Reply:
In fact, the animation of a specific station has an arbitrary number of stages, for example, ground stations have none at all. And it is read from the tags in the model, although there is some hard-coded logic that adds intermediate "locators". This is old code and it works well with current models.
I think it will be easy enough to turn this logic into a separate dynamic list when needed. 
External beacons are stored separately in `DockStage::APPROACH1` and `DockStage::APPROACH2`. I think that conceptually they should be located separately, so that, for example, they do not rotate with the station.